infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -38,14 +38,26 @@ public void sealServerWithEpoch(long epoch) {
 
     public abstract boolean isServerReadyToHandleMsg(CorfuMsg msg);
 
+    /**
+     * A stub that handlers can override to manage their threading, otherwise
+     * the requests will be executed on the IO threads
+     * @param msg
+     * @param ctx
+     * @param r
+     */
+    protected void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        getHandler().handle(msg, ctx, r);
+    }
+
     /**
      * Handle a incoming Netty message.
      *
      * @param msg An incoming message.
      * @param ctx The channel handler context.
      * @param r   The router that took in the message.
      */
-    public void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    // todo(Maithem): make sure this prevents subclasses from overriding the method
```

Comment:
This TODO seems to be done and can be removed as you made the method final.

Reply:
Done.